Side drawer

raoulkent opened the pull request:
This merge adds a very basic sidebar to the app.
It also includes several changes to how the uploading and reading of data is performed.

Furthermore it has many naming refactors, as well as removing unused- and organising existing imports.

raoulkent committed:
Wrap, and add, single checkbox

raoulkent committed:
Fix previous commit by using pre-commit

linneahallin committed:
Add reset and hide for no_operator_checkbox

raoulkent committed:
Refactor naming of widgets, components and their methods

Rename to municipalities

raoulkent committed:
Add SideBar component

Implement the basics of the sidebar with experimental bootstrap styling.

raoulkent committed:
Move filters in tabs into the sidebar component

raoulkent committed:
Refactor filter components to be FormGroups

Refactor op_code_selection and op_time_slider, to standardise filter components.

raoulkent committed:
Naming refactor of most components

Remove ..widget from names, to increase readability.
Add constants to replace some components use of strings.

raoulkent committed:
Create first pass of adding the sidebar, with callback, to main

Requires refactor of main, by extracting code currently in main to individual
components.
Such as components for content as well as the custom css styles.

raoulkent committed:
Make initial custom sidebar with toggle button

Sidebar lacks styling

raoulkent committed:
Minor restyling

raoulkent committed:
Clean up main.py

linneahallin committed:
Change px to em

raoulkent committed:
Merge branch 'main' into side_drawer

raoulkent committed:
Refactor, mainly naming

linneahallin committed:
Change to collapse

davidjohnsson96 committed:
Refactor upload, add warning widget

raoulkent committed:
Merge branch 'David' into side_drawer

davidjohnsson96 reviewed (commented):
Approved by me (David), partially contains my branch so someone else should verify and merge

davidjohnsson96 committed:
small refactor, change interval of op-time, add op codes based on data

davidjohnsson96 committed:
Merge branch 'operator' into side_drawer

# Conflicts:
#	components/operator.py

davidjohnsson96 reviewed (dismissed):
Noticed breaking of naming conventions for functions/methods "getComponent". I can fix it

davidjohnsson96 committed:
merge operator into side_drawer branch, refactor code

raoulkent commented:
> Noticed breaking of naming conventions for functions/methods "getComponent". I can fix it

Ah right, it does indeed.
I think the 'tab selection' component is excess now as well.

GideonJagen commented on components/anestesi.py:
```diff
@@ -4,36 +4,36 @@
 
 
-class AnestesiWidget:
+class Anaesthetic:
```
Anesthesia instead of Anaesthetic.

"Anaesthesia means "loss of sensation". Medicines that cause anaesthesia are called anaesthetics."

GideonJagen reviewed (changes requested):
Anesthesia instead of Anaesthetic in anestesi.py.
"Anaesthesia means "loss of sensation". Medicines that cause anaesthesia are called anaesthetics."

Should change to name of following python files to match class names: 
- anestesi.py -> anesthesia.py
- number_patients.py -> patient_count.py
- op_code_selection.py -> op_code.py
- op_time_slider.py -> op_time.py

raoulkent committed:
Refactoring filenames and reset functions

raoulkent reviewed (commented):
> Anesthesia instead of Anaesthetic in anestesi.py.
> "Anaesthesia means "loss of sensation". Medicines that cause anaesthesia are called anaesthetics."
> 
> Should change to name of following python files to match class names:
> 
> * anestesi.py -> anesthesia.py
> * number_patients.py -> patient_count.py
> * op_code_selection.py -> op_code.py
> * op_time_slider.py -> op_time.py

Fixed in 21a580a

raoulkent committed:
Refactor rename most callbacks for components to add_callback()

GideonJagen reviewed (changes requested):
- Update the callback adding in main.py (line 61 - 73) from `add_class_callback(app)` to `add_callback(app)`
- Change name of Upload to FileUpload? (Not sure what was decided)

raoulkent committed:
Fix callbacks and optimize imports

Fix adding of callbacks in main.py to use the correct functions.
Optimize imports for practically all python files.
Change naming from Upload to FileUpload

raoulkent committed:
Rename to file_upload and care_type

GideonJagen commented on components/operator.py:
```diff
@@ -17,91 +15,105 @@ class OperatorWidget:
 
     @staticmethod
-    def _operator_radiobuttons():
+    def _operator_radioitems():
```
Either change this back to radiobuttons or change line 16 in care_type.py and line 19 in municipalities.py to radioitems (and any other places where it's wrong)

raoulkent replied:
Fixed in 4c8a45ad45748c7f274e83dbcd0c06abcf030bef

GideonJagen commented on components/sidebar.py:
```diff
@@ -0,0 +1,82 @@
+from components.statistics_code import StatisticsCode
+
+
+class SideBar:
```
change class name to Sidebar or change file name to side_bar.py

raoulkent replied:
Fixed in 4189132

GideonJagen commented on data_handler.py:
```diff
@@ -1,12 +1,13 @@
+import pandas as pd
 
 
 class DataFilterer:
```
Class name does not match file name

raoulkent replied:
Fixed in 4189132. Left the name as `data_handling` as the file has two classes, with that name a descriptor of their grouping.

GideonJagen reviewed (changes requested):
Added a few comments about name changes.

Have tried running it and could not find any apparent bugs. 

Should be good to merge after next fix

raoulkent committed:
Refactor radiobuttons to radioitems

raoulkent committed:
Rename SideBar->Sidebar & rename data_handler->data_handling

GideonJagen reviewed (approved):
Looks good now.
